File: frontend/src/db/schemas/product.ts
import type { RxJsonSchema } from 'rxdb'

export interface ProductDoc {
  id: string
  categoryId: string
  name: string
  price: number
  stock: number | null
  barcode: string | null
  imageDataUrl: string | null
  isActive: boolean
  sortOrder: number
  createdAt: number
  updatedAt: number
}

export const productSchema: RxJsonSchema<ProductDoc> = {
  version: 1,
  primaryKey: 'id',
  type: 'object',
  properties: {
    id: { type: 'string', maxLength: 36 },
    categoryId: { type: 'string', maxLength: 36 },
    name: { type: 'string' },
    price: { type: 'number', multipleOf: 1, minimum: 0, maximum: 9999999 },
    stock: { type: ['number', 'null'] },
    barcode: { type: ['string', 'null'] },
    imageDataUrl: { type: ['string', 'null'] },
    isActive: { type: 'boolean' },
    sortOrder: { type: 'number', multipleOf: 1, minimum: 0, maximum: 10000 },
    createdAt: { type: 'number', multipleOf: 1, minimum: 0, maximum: 99999999999999 },
    updatedAt: { type: 'number', multipleOf: 1, minimum: 0, maximum: 99999999999999 },
  },
  required: ['id', 'categoryId', 'name', 'price', 'isActive', 'sortOrder', 'createdAt', 'updatedAt'],
  indexes: ['categoryId', 'sortOrder'],
}

File: backend/app/models/product.py
import uuid

from sqlalchemy import String, Integer, Boolean, ForeignKey, BigInteger
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Product(Base):
    __tablename__ = "products"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    exhibition_id: Mapped[str] = mapped_column(String(36), ForeignKey("exhibitions.id"))
    category_id: Mapped[str] = mapped_column(String(36), ForeignKey("categories.id"))
    name: Mapped[str] = mapped_column(String(255))
    price: Mapped[int] = mapped_column(Integer)
    stock: Mapped[int | None] = mapped_column(Integer, nullable=True)
    barcode: Mapped[str | None] = mapped_column(String(255), nullable=True, default=None)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[int] = mapped_column(BigInteger)
    updated_at: Mapped[int] = mapped_column(BigInteger)
